### ISSUE TITLE
Will DepthwiseConv2D support non-square Strides in the future? Ex: `strides=(1,3)`

### ISSUE THREAD

**Issue by Raukk** [tensorflow/tensorflow#33005]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 
2.0.0, 1.14.0
- Are you willing to contribute it (Yes/No): 
No, sorry, that is well outside my expertise. 

**Describe the feature and the current behavior/state.**
Will DepthwiseConv2D support non-square Strides in the future? 
This would make it consistent with the majority of other layers that accept the strides argument and can take non-square strides. Ex: `strides=(1,3)`

Or at minimum, please update the documentation to specify this limitation.
https://www.tensorflow.org/api_docs/python/tf/keras/layers/DepthwiseConv2D
"strides: An integer or tuple/list of 2 integers, specifying the strides of the convolution along the height and width."

depthwise_conv_op
"Current implementation only supports equal length strides in the row and column dimensions."
https://github.com/tensorflow/tensorflow/blob/242a42139513aa5903743c37c0ef15ed00f93fed/tensorflow/core/kernels/depthwise_conv_op.cc#L288

**Will this change the current api? How?**
No, there is already a strides option that should be able to take non-square shaped strides. This would actually make it more consistent with the rest of the API.

**Who will benefit with this feature?**
Bob will.

**Any Other info.**
I'm using the `tf.keras.layers.DepthwiseConv2D`  implementation.

2.0.0:
`tensorflow.python.framework.errors_impl.InvalidArgumentError: Current implementation only supports equal length strides in the row and column dimensions. [Op:DepthwiseConv2dNative] name: model/depthwise_conv2d/depthwise/`

1.14.0:
`tensorflow.python.framework.errors_impl.InvalidArgumentError: Current implementation only supports equal length strides in the row and column dimensions.
         [[{{node depthwise_conv2d/depthwise}}]]`

Simple example:
`tf.keras.layers.DepthwiseConv2D((3, 3), strides=(1,2))`

**Comment by jvishnuvardhan** [comment]
@Raukk Can you please mention any use case for this kind of feature? Thank!

**Comment by Raukk** [comment]
@jvishnuvardhan In my understanding, I think the same use cases for `Conv2D` would apply to a `DepthwiseConv2D` and `SeparableConv2D`, since they are essentially a drop in replacement for a `Conv2D` (that do different math under the hood). If you know of a reason why the `Conv2D` Use-cases would not apply to `DepthwiseConv2D` or `SeparableConv2D` then I can look for Specific Use-cases.

I'll link to a set of stupid Colab Notebooks that show it working with `Conv2D` but failing if I replace any of the Conv2D layers with either `SeparableConv2D` or `DepthwiseConv2D` beacuse of the stride.

In general, I expect the layer arguments in the Keras API to be consistent across all layers that use them (except where noted by the documentation). A few examples of these arguments are `kernel_size`, `strides`, `padding`, `data_format`, and `dilation_rate` which I expect to have consistent usage, formatting, and effect. In this case, the usage and behavior of `tf.keras.layers.Conv2D(filters, kernel_size, strides=(1, 2))` to match `tf.keras.layers.SeparableConv2D(filters, kernel_size, strides=(1, 2))` and `tf.keras.layers.DepthwiseConv2D(kernel_size, strides=(1, 2))`


I would totally accept that the structure of the calculations or the optimizations make it impossible to implement this functionality. At which point I would ask that it be added to the documentation that this case is not implemented. I'm sure you'd agree that it's really frustrating when something isn't implemented, but the documentation implies that it is implemented. All three layers (Conv2D, SeparableConv2D, DepthwiseConv2D) say the same wording in the online documentation:
"strides: An integer or tuple/list of 2 integers, specifying the strides of the convolution along the height and width."
https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D
https://www.tensorflow.org/api_docs/python/tf/keras/layers/SeparableConv2D
https://www.tensorflow.org/api_docs/python/tf/keras/layers/DepthwiseConv2D

Colab Links:
https://colab.research.google.com/gist/Raukk/288b0889fe4f7200937af6914b7a0fe1/stride_standardconv_example_tf2.ipynb
https://colab.research.google.com/gist/Raukk/7f494ebbf99fafe24a187c37fc4483b4/stride_separableconv_example_tf2.ipynb
https://colab.research.google.com/gist/Raukk/774b026b28365328d0bbb1acebd92707/stride_depthwiseconv_example_tf2.ipynb
Please let me know if you have an issue with these links.

**Comment by michaelowenliu** [comment]
Yes! I agree with you. The document indicates it has been implemented already, but when I prepared everything and started training, it shows the 'not support' errors, which is frustrating.

**Comment by msdrigg** [comment]
I would also like to see some sort of response to this.

**Comment by Maximellerbach** [comment]
Still waiting for this feature to be added ;)

**Comment by aidansmyth95** [comment]
Likewise, waiting for this feature to be added (hopefully for TF >= 1.14)

**Comment by Luux** [comment]
Just stumbled upon this issue. I wanted to implement an EfficientNet that uses (1, 2)-strides. However, this does not seem to be possible with the current implementation of Depthwise Conv?

**Comment by arpadfodor** [comment]
The issue still persists with TF 2.5.0, there is no sign of any progress at all. Please, it would be awesome having non-squared strides, or at least clarifying the documentation @tensorflowbutler.

**Comment by alexmil2019** [comment]
I have checked on nightly but still has the same issue. can you fix this?
``` 
tensorflow.python.framework.errors_impl.InvalidArgumentError:  Current implementation only supports equal length strides in the row and column dimensions.
         [[{{node model/conv/StatefulPartitionedCall/StatefulPartitionedCall/dw_conv2d_1/depthwise}}]] [Op:__inference_predict_function_43730]```

**Comment by sushreebarsa** [comment]
@Raukk 
In order to expedite the trouble-shooting process here,Could you please fill the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose),
Thank you!

**Comment by sachinprasadhs** [comment]
Hi, The Arg details has been updated in the document, please refer the document here  https://www.tensorflow.org/versions/r2.12/api_docs/python/tf/keras/layers/DepthwiseConv2D#args 
and feel free to close the issue. Thanks!
